code-challenges.test.js

```diff
@@ -15,27 +15,98 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
-
+//1.function deCode()
+//2. converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+//3. returns message
+//  ReferenceError: deCode is not defined
+
+//       28 |     const secretCodeWord3 = "Eccentric"
+//       29 |     // Expected output: "3cc3ntr1c" 
+//     > 30 |     expect(deCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+//          |     ^
+//       31 |     expect(deCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//       32 |     expect(deCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+//       33 |   })
+
+//       at Object.<anonymous> (code-challenges.test.js:30:5)
+
+// Test Suites: 1 failed, 1 total
+// Tests:    
+// Failed!!
+describe("deCode", () => {
+ 
+  it("takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+     const secretCodeWord1 = "Lackadaisical"
+    // Expected output: "L4ck4d41s1c4l"
+    const secretCodeWord2 = "Gobbledygook"
+    // Expected output: "G0bbl3dyg00k"
+    const secretCodeWord3 = "Eccentric"
+    // Expected output: "3cc3ntr1c" 
+    expect(deCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    expect(deCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    expect(deCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+  })
+})
+const deCode =(str)=>{
+    newStr1 =str.replaceAll(/a/gi,"4")
+    //converts a =>4
```

Comment:
While this is some pseudocode, I would really like to see your pseudocode breaking down this entire problem step by step as well.  Pseudocode is generally going to be written out before your start even writing your function and is your roadmap when building the function.  It will also mention the different methods being used and what they will do for you inside your function.